proposals/nnnn-distributed-actors.md

```diff
@@ -0,0 +1,1621 @@
+
+#### Initialization
+Distributed actors and functions are largely a set of a rules that enforce location transparency. In order for distributed actors to actually perform any remote communication, it is necessary to initialize them with an actual implementation of an `ActorTransport`, which is a protocol that is implementable by library writers. A *transport* handles all of the connection details that underpin distributed actors. An actor transport can take advantage of any networking, cross-process, or even in-memory approach to communicate between instances, as long as it conforms to the protocol correctly.
+
+There are two ways to initialize a distributed actor, but in any case, a distributed actor is always associated with some `ActorTransport`. To create a new **local** instance of a distributed actor, call the actor's `init` as usual. All non-delegating initializers, which fully-initialize the distributed actor, are required to accept exactly one argument that conforms to `ActorTransport`:
+
+```swift
+distributed actor Capybara { 
+  var name: String
+  
+  // ✅ ok. exactly one transport param for non-delegating initializer.
+  init(named name: String, using: ActorTransport) {
+    self.name = name
+  }
+
+  // ✅ ok. no transport param, but it's a delegating initializer.
+  convenience init() {
+    self.init(named: "Happybara", using: Defaults.IPoAC)
+  }
+
+  // ❌ error, too many transport params for a non-delegating init.
+  init(main: ActorTransport, alt: ActorTransport) {
+    self.name = "Sleepybara"
+  }
+  
+  // ❌ error, no transports for a non-delegating init.
+  init(withName name: String) {
+    self.name = name
+  }
+}
+```
+
+Swift will automatically associate a `Capybara` instance with the transport passed to `init(named:using:)`, without the programmer explicitly using the transport argument. This is done, instead of requiring a specific `let`-bound property to be initialized, because the `ActorTransport` is specially stored in the instance, depending on whether it represents a remote or local instance of the actor.
+
+A **remote** instance can be only be resolved if the programmer also has the instance's `ActorIdentity`. An `ActorIdentity` is a protocol that specifies the minimum capabilities of a durable, unique identifier associated with a distributed actor instance. All distributed actors have two computed properties that are accessible from anywhere, and a static `resolve` function:
+
+```swift
+public protocol DistributedActor: Sendable, Codable, Identifiable, Hashable {
+
+  /// Logical identity of this distributed actor.
+  ///
+  /// Many distributed actor references may be pointing at, logically, the same actor.
+  /// For example, calling `resolve(id:using:)` multiple times, is not guaranteed
```

Comment:
The doc is inconsistent about whether the first param of `resolve()` has a label, won't call out every one.  I think the `resolve(_:using:)` version reads fine.